src/Microsoft.VisualStudio.LanguageServices.Razor/ProjectSystem/RazorProjectHostBase.cs

```diff
@@ -43,6 +45,7 @@ internal abstract class RazorProjectHostBase : OnceInitializedOnceDisposedAsync,
 
             _lock = new AsyncSemaphore(initialCount: 1);
             _currentDocuments = new Dictionary<string, HostDocument>(FilePathComparer.Instance);
+            _hostProjectId = ProjectId.CreateNewId();
```

Comment:
What value does this add by being persistent? What would happen if you created the ID on 'project add' and then threw it away on 'project remove'?

My concern is about reusing the same ID across say, a project reload (think about unload+reload from VS). That doesn't seem good, but I'm not sure it matters.

Reply:
Hu? if you unload a project this does get re-created; CPS calls dispose on our project hosts and re-instantiates them on reload.

Reply:
ok dope